src/views/virtual_list.rs

```diff
@@ -120,18 +121,25 @@ where
         },
         move |(_, e)| key_fn(e),
         move |(index, e)| {
-            Item {
-                data: ViewData::new(Id::next()),
-                selection,
-                index,
-                child: Box::new(view_fn(e)),
-            }
-            .on_click_stop(move |_| {
-                if selection.get_untracked() != Some(index) {
-                    selection.set(Some(index))
+            container(
+                Item {
+                    data: ViewData::new(Id::next()),
+                    selection,
+                    index,
+                    child: view_fn(e).any(),
                 }
+                .on_click_stop(move |_| {
+                    if selection.get_untracked() != Some(index) {
+                        selection.set(Some(index))
+                    }
+                })
+                .style(|s| s.width_full()),
+            )
+            .class(ListItemClass)
+            .style(move |s| match direction {
+                VirtualDirection::Horizontal => s.flex_row(),
```

Comment:
Then this would become `height_full()` and `width_full()` instead of `flex_row()` and `flex_col()` on `Item` instead.